### DISCUSSION TITLE
Beginner Question

### DISCUSSION THREAD

**Discussion by rishujam** [kiwix/kiwix-android#2752] [Q&A] [answered]
Hello, I want to learn more in android by contributing, I am new to open source can I get some beginner friendly tasks in this project. 
I saw the code and readme file and its quite understandable for me.

**Comment by gouri-panda** [comment, marked as answer]
Hello! @rishujam Find good first issue and choose whatever you like.

**Comment by rishujam** [comment]
I have a basic question that why we are making packages outside the app package like the core package and many other.

**Reply by kelson42** [reply]
Which package do you exactly refer to?

**Reply by rishujam** [reply]
Core package, Like in android view we only see the app package in android studio
So can i say that core package is just a library

**Reply by rishujam** [reply]
Or are these just test code because when I see usage of some files in core package it just refers to test package

**Reply by cdhiraj40** [reply]
@rishujam I think you meant different modules?
eg.
![image](https://user-images.githubusercontent.com/75211982/146199956-1b35cc1b-3e8e-4eb8-881a-d2348ffc6390.png)

**Reply by cdhiraj40** [reply]
I think reasons for using different modules are that:
1. it helps to divide the project into discrete units of functionality
2. if I'm not wrong it also helps in decreasing the build process [more information](https://www.freecodecamp.org/news/how-modularisation-affects-build-time-of-an-android-application-43a984ce9968/)

here's a [link](https://developer.android.com/studio/projects/index.html#ApplicationModules) explaining modules.

**Reply by rishujam** [reply]
thanks

**Comment by rishujam** [comment]
@gouri-panda Will kiwix participate in gsoc 2023?

**Comment by gouri-panda** [comment]
@rishujam We will apply like every year but google decides on slot allocation. Not sure about the projects yet, they'll be decided in February.

**Comment by rishujam** [comment]
okay Thanks
